MAUI/DataGrid/row-drag-and-drop.md

```diff
@@ -0,0 +1,402 @@
+{
+    var totalHeight = dataGrid.RowColumnIndexToPoint(new RowColumnIndex(viewModel.Orders.Count, 0)).Y + this.dataGrid.RowHeight;
+    if (e.DraggingAction == DataGridDragAction.DragEnded)
+    {
+        if (Math.Ceiling(e.Position.Y + (dataGrid.RowHeight / 2)) > totalHeight && e.To == viewModel.Orders.Count)
+        {
+            //Will hit if the row is dropped at the last position                 
+            DisplayAlert("RowDragAndDrop info", "The row is dropped at the last position", "OK");
+        }
+    }
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+
+## Updating summaries when dragging and dropping a row between groups
+
+Grouping and summarization of items in the data grid are determined by the group key. When an item is dragged from one group to another, the group key of the dragged item will differ from the group key of the items in the dropped group. Therefore, by default, the summaries will not be updated. This reflects the actual behavior of this control.
+
+Therefore, to update the summaries when a row is dragged and dropped between groups, call the `UpdateCaptionSummaries` and `Refresh` methods in the `QueryRowDragging` event.
+
+{% tabs %}
+{% highlight c#%}
+public partial class MainPage : ContentPage
+{
+    private SfDataGrid dataGrid;
+    private ViewModel viewModel;
+    public MainPage()
+    {
+         InitializeComponent();
+         dataGrid = new SfDataGrid();
+         viewModel = new OrderRepo();
+         dataGrid.ItemsSource = viewModel.Orders;
+         dataGrid.QueryRowDragging += DataGrid_QueryRowDragging;
+         this.Content = dataGrid;
+    }
+   private void DataGrid_QueryRowDragging(object? sender, Syncfusion.Maui.DataGrid.DataGridQueryRowDraggingEventArgs e)
+    {
+         if (e.DraggingAction == DataGridDragAction.DragEnded)
+         {
+             // Delay is given for refreshing the view.
+             await Task.Delay(100);                
+             this.dataGrid.View.TopLevelGroup.UpdateCaptionSummaries();
+             this.dataGrid.View.Refresh();
+         }
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+
+## Cancel auto scrolling 
+
+The vertical auto-scrolling of the `SfDataGrid` during row drag-and-drop can be canceled using the `CanAutoScroll` argument in the `QueryRowDragging` event handler.
+
+{% tabs %}
+{% highlight c# %}
+
+dataGrid.QueryRowDragging += DataGrid_QueryRowDragging;
+
+private async void DataGrid_QueryRowDragging(object? sender, Syncfusion.Maui.DataGrid.DataGridQueryRowDraggingEventArgs e)
+{
+   e.CanAutoScroll = true;
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+
+## RowDragAndDrop Style
+### RowDraggingIndicatorLineColor
+
+The `RowDraggingIndicatorLineColor` property in the `SfDataGrid` is used to customize the color of the line that appears during a drag-and-drop operation when you move rows within the dataGrid. This line serves as a visual indicator, showing where the row will be dropped if the user releases the drag at that point.
```

Comment:
The row view’s background color, text color, and indicator color can be customized using the RowDragViewBackgroundColor, RowDragViewTextColor, and RowDraggingIndicatorLineColor properties.

// Remove the three topics and provide only one code snippet for all the properties here and provide image also.

Reply:
Removed and updated